### DISCUSSION TITLE
Best approach for online learning

### DISCUSSION THREAD

**Discussion by bmsan** [ourownstory/neural_prophet#697] [Q&A - forecasting best practices] [answered]
Hello,

What would be the best approach to incorporate new data as it becomes available when doing future predictions?

Let's say I'm interested in the next value of a timeseries. I can train the model based on the history and predict the next sample.(N+1)

But once the new sample becomes available I would want to do the prediction again(for the next future sample N+2) but also incorporate the information held by N+1.

1. Do I need to retrain the algorithm from scratch ?

2. Is there a way to fine-tune the already trained algorithm with the new data?

3. Assuming I need to retrain from scratch, can I carry over some of the hyperparmeters from the previous trained model and use them ? (In order for the retraining to be faster)

Thanks

**Comment by ourownstory** [comment, marked as answer]
Hi @bmsan ,

Currently we do not have a proper online learning approach as we are catering to wide set of applications (hyperparameters would be a challenge). So, 2) No 

However, if you fit a model, and then later have new data, you can simply update your input dataframe, and then call predict again.
If you are using Autoregression or lagged regressors, this will make use of the new data.

With time, your data may differ from the training data, and thus you can retrain the model with the updated dataset. -> 1) Yes
And 3) Yes, you can just use the same hyperparameters in your retraining. At least until your data has changed significantly (like once a year or quarter you may want to retune).
